quantestpy/assertion/assert_unary_iteration.py

```diff
@@ -0,0 +1,225 @@
+import copy
+import re
+import sys
+from typing import Union
+
+import numpy as np
+
+from quantestpy.assertion.assert_circuit_equivalent_to_output_qubit_state \
+    import PauliCircuitDrawerColorErrorQubit
+from quantestpy.converter.converter_to_quantestpy_circuit import \
+    cvt_input_circuit_to_quantestpy_circuit
+from quantestpy.exceptions import QuantestPyAssertionError, QuantestPyError
+from quantestpy.simulator.pauli_circuit import (
+    PauliCircuit, cvt_quantestpy_circuit_to_pauli_circuit)
+
+
+def _is_a_in_b(a: list, b: list) -> bool:
+    for i in a:
+        if i in b:
+            return True
+    return False
+
+
+def _assert_internal_replacing_gates_in_system_reg_with_x_gates(
```

Comment:
It might be a good idea to change the name of this method because this assert includes not just replacing but executing and checking. For example,
- `_assert_qubit_value_in_system_reg_with gates_replaced_by_x`
- etc...

Reply:
Done! 👉 https://github.com/QuantestPy/quantestpy/pull/187/commits/ad21166478bc4908f1777f4f8b889b473f96dd4b